packages/cletus/src/operations/internet.tsx

```diff
@@ -0,0 +1,242 @@
+import puppeteer from 'puppeteer';
+import { abbreviate } from '../common';
+import { operationOf } from './types';
+import { renderOperation } from '../helpers/render';
+
+// ============================================================================
+// Web Search
+// ============================================================================
+
+export const web_search = operationOf<
+  { query: string; maxResults?: number },
+  { query: string; results: Array<{ title: string; url: string; content: string }> }
+>({
+  mode: 'read',
+  signature: 'web_search(query: string, maxResults?: number)',
+  status: (input) => `Searching web: ${abbreviate(input.query, 35)}`,
+  analyze: async (input, { config }) => {
+    const tavilyConfig = config.getData().providers.tavily;
+    
+    if (!tavilyConfig?.apiKey) {
+      return {
+        analysis: 'This would fail - Tavily API key not configured. Please add it in Settings > Manage Providers.',
+        doable: false,
+      };
+    }
+
+    return {
+      analysis: `This will search the web for "${input.query}" and return up to ${input.maxResults || 5} results.`,
+      doable: true,
+    };
+  },
+  do: async (input, { config }) => {
+    const tavilyConfig = config.getData().providers.tavily;
+    
+    if (!tavilyConfig?.apiKey) {
+      throw new Error('Tavily API key not configured');
+    }
+
+    // Dynamically import tavily to avoid issues if not installed
+    const { tavily } = await import('@tavily/core');
+    const client = tavily({ apiKey: tavilyConfig.apiKey });
+
+    const response = await client.search(input.query, {
+      maxResults: input.maxResults || 5,
+    });
+
+    const results = response.results.map((r: any) => ({
+      title: r.title || '',
+      url: r.url || '',
+      content: r.content || '',
+    }));
+
+    return {
+      query: input.query,
+      results,
+    };
+  },
+  render: (op) => renderOperation(
+    op,
+    `WebSearch("${abbreviate(op.input.query, 30)}")`,
+    (op) => {
+      if (op.output) {
+        return `Found ${op.output.results.length} result${op.output.results.length !== 1 ? 's' : ''}`;
+      }
+      return null;
+    }
+  ),
+});
+
+// ============================================================================
+// Web Get Page
+// ============================================================================
+
+export const web_get_page = operationOf<
+  { url: string; type: 'html' | 'text'; regex?: string; surrounding?: number; lineStart?: number; lineEnd?: number },
+  { url: string; totalLines: number; content: string; matches?: string[] }
+>({
+  mode: 'read',
+  signature: 'web_get_page(url: string, type: "html" | "text", regex?, surrounding?, lineStart?, lineEnd?)',
+  status: (input) => `Fetching page: ${abbreviate(input.url, 35)}`,
+  analyze: async (input) => {
+    const parts: string[] = [`This will fetch the ${input.type} content from "${input.url}"`];
+    
+    if (input.regex) {
```

Comment:
Validate regex - if it's invalid return { analysis: 'with error', doable: false }